### PR TITLE
Setting custom n-handler-threads is not supported in per-cpu mode

### LINKED ISSUE #3842: n-handler-threads is overridden, causing ovs-node pod to OOM after applying ovnkube-node daemonset
I'm installing [ovn-kubernetes](https://github.com/ovn-org/ovn-kubernetes/tree/master) and see my ovs-node pod OOM after applying the ovnkube-node daemonset. This is running on an instance with 96 cores.

When starting the ovs-node, n-handler-threads is set to 10 per the [ovnkube.sh](https://github.com/ovn-org/ovn-kubernetes/blob/master/dist/images/ovnkube.sh#L668-L671) script (nproc is 96):
```
if [[ $(nproc) -gt 12 ]]; then
  ovs-vsctl --no-wait set Open_vSwitch . other_config:n-revalidator-threads=4
  ovs-vsctl --no-wait set Open_vSwitch . other_config:n-handler-threads=10
fi
```
Confirmed on my machine:
```
[root@dev-vm00001 ~]# ovs-vsctl get Open_vSwitch . other_config
{n-handler-threads="10", n-revalidator-threads="4"}
```
After applying ovnkube-node daemonset, ovs-node logs:
```
2023-08-18T22:44:27.909Z|00034|ofproto_dpif_upcall|INFO|Overriding n-handler-threads to 96, setting n-revalidator-threads to 4
2023-08-18T22:44:27.909Z|00035|ofproto_dpif_upcall|INFO|Starting 100 threads
2023-08-18T22:44:30.140Z|00002|daemon_unix(monitor)|INFO|pid 70686 died, killed (signal 9), exiting
OVS seems to have crashed, exiting
```
How do I stop n-handler threads from being overridden?

### PR DESCRIPTION
**Pull request by arghosh93**
**- What this PR does and why is it needed**

Function[1] sets n_handler to maximum number of CPUs available on the node and in turn overrides n-handler-threads while setting[2] up threads. So, right now there is no point in setting n-handler-threads to 10 when number of CPUs are more than 12. Limiting n-revalidator-threads to 4 also not at per with the best practices in per-cpu mode.
[1] - https://github.com/openvswitch/ovs/blob/master/lib/dpif-netlink.c#L2570-L2591
[2] - https://github.com/openvswitch/ovs/blob/master/ofproto/ofproto-dpif-upcall.c#L650

**- Special notes for reviewers**

Currently n-revalidator-threads can be limited by setting n-revalidator-threads other_config but that is not suitable in per-cpu mode and its best to create default(n-handler-threads/4+1) number of threads for revalidator.
closes #3842

**- How to verify it**
Check ovs-vswitchd log

**- Description for the changelog**
- Removing code block to disable setting custom n-handler-threads and n-revalidator-threads when number of CPUs are more than 12.
- Adding a warning message to notify high memory allocation due to creation of high number of threads in a node with high number of CPUs.
- Increasing CPU and memory limit of ovs-node daemonset to prevent OOM caused by high memory allocation per threads.
- Limiting stack size to 2M as per below commit on Openvswitch
  https://github.com/openvswitch/ovs/commit/b82a90e266e1246fe2973db97c95df22558174ea

### CODE REVIEW

**Comment by tssurya** [comment]
flakes unrelated to this change;
https://github.com/ovn-org/ovn-kubernetes/actions/runs/6277731189/job/17050832921?pr=3921
https://github.com/ovn-org/ovn-kubernetes/actions/runs/6277731189/job/17050833310?pr=3921
saving for history/further debugging.

**Comment by amorenoz** [comment]
From OVS's perspective this makes much more sense than the previous code. Thanks @arghosh93.

File: dist/images/ovnkube.sh [hunk 1]
```diff
@@ -672,6 +672,11 @@ ovs-server() {
     echo "Warning: Higher memory allocation by ovs-vswitchd is expected due to high number of n-handler-threads and n-revalidator-threads"
   fi
 
+  # Reduce stack size to 2M from default 8M as per below commit on Openvswitch
+  # https://github.com/openvswitch/ovs/commit/b82a90e266e1246fe2973db97c95df22558174ea
+  # added while troubleshooting on https://bugzilla.redhat.com/show_bug.cgi?id=1572797 
+  ulimit -s 2048
```
Comment (martinkennelly) [inline dist/images/ovnkube.sh:678]: I dont see this being applied in my KinD cluster within `ovs-node-*` pod:

```shell
$oc -n ovn-kubernetes rsh ovs-node-j6wc9
sh-5.2# ulimit -s
8192
```

File: dist/templates/ovs-node.yaml.j2 [hunk 2]
```diff
@@ -86,8 +86,8 @@ spec:
             cpu: 100m
             memory: 300Mi
           limits:
-            cpu: 200m
-            memory: 400Mi
+            cpu: 500m
+            memory: 500Mi
```
Comment (martinkennelly) [inline dist/templates/ovs-node.yaml.j2:90]: Wont we have the same issue again with 256 cores? Why not just remove the memory limit?

```
524288÷2048=256
```

**Comment by martinkennelly** [comment]
Excellent research done here - it was easy for me to get up to speed to understand the problem and solution! Thanks @arghosh93

File: dist/images/ovnkube.sh (hunk 1 above)
Comment (arghosh93) [reply dist/images/ovnkube.sh:678]: My bad, I should have set ulimit before starting ovs-vswitchd. I will modify and recommit.

 `ulimit -s` gives session specific output. I am verifying the fix by checking content of /proc/$(pidof ovs-vswitchd)/limits inside ovs-node POD.

File: dist/templates/ovs-node.yaml.j2 (hunk 2 above)
Comment (arghosh93) [reply dist/templates/ovs-node.yaml.j2:90]: Yes, the issue would reoccur if the node have 256 cores. is it okay to get rid of the limit completely?

Comment (martinkennelly) [reply dist/templates/ovs-node.yaml.j2:90]: *I* think its fine for upstream but Ill let the maintainers decide. Wait until they weigh in.

Comment (martinkennelly) [reply dist/templates/ovs-node.yaml.j2:90]: 256 cores isnt that far away anymore :)

**Comment by martinkennelly** [comment]
you can remove the merge commit also: `Merge branch 'master' into upstream_3842_n-handler-threads`

File: dist/templates/ovs-node.yaml.j2 (hunk 2 above)
Comment (trozet) [reply dist/templates/ovs-node.yaml.j2:90]: @girishmg wdyt?

**Comment by martinkennelly** [comment]
/lgtm

build failed because of known flake tracked here: #3887 - see last comment
retesting

File: dist/templates/ovs-node.yaml.j2 (hunk 2 above)
Comment (martinkennelly) [reply dist/templates/ovs-node.yaml.j2:90]: @arghosh93 did you check if we need to adjust anything downstream within OCP?

Comment (arghosh93) [reply dist/templates/ovs-node.yaml.j2:90]: @martinkennelly We run ovs-vswitchd as a systemd unit and add dropins[1] along with systemd unit[2] which comes with Openvswitch on RHEL. LimitSTACK is set in default ovs-vswitchd systemd unit[2].

[1] - https://github.com/openshift/machine-config-operator/blob/c74052cba27ebeb11a697d579ce99c4f47e47a0f/templates/common/_base/units/ovs-vswitchd.service.yaml
[2] - https://github.com/openvswitch/ovs/blob/master/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in#L18

**Review by martinkennelly — APPROVED** [review]

File: dist/templates/ovs-node.yaml.j2 (hunk 2 above)
Comment (martinkennelly) [reply dist/templates/ovs-node.yaml.j2:90]: Sweet! Thanks @arghosh93 
```
LimitSTACK= | ulimit -s | Bytes
```
https://www.freedesktop.org/software/systemd/man/systemd.exec.html

Comment (trozet) [reply dist/templates/ovs-node.yaml.j2:90]: haven't heard back from @girishmg yet, and this PR is not removing the limits. I think we can handle removing the limits in a follow up PR, and merge what we have here.

**Review by trozet — APPROVED** [review]

**Pull request merged**